### PR TITLE
Implement LeaderboardScopeSelector

### PR DESCRIPTION
**Pull request by EVAST9919**
Split from https://github.com/ppy/osu/pull/5451

### CODE REVIEW

**Comment by peppy** [comment]
Why is this not using `OsuTabControl`, which already implements this tab design? We shouldn't be implementing local variants.

See `HeaderTabControl` for instance. It's also used at song select and other places.

File: osu.Game/Overlays/BeatmapSet/LeaderboardScopeSelector.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colours)
+            {
+                box.Colour = colours.Blue;
+            }
+
+            protected override bool OnHover(HoverEvent e)
+            {
+                Text.FadeColour(Color4.LightSkyBlue);
```
Comment (peppy) [inline osu.Game/Overlays/BeatmapSet/LeaderboardScopeSelector.cs]: wrong colour. we never use color4.

**Review by peppy — CHANGES_REQUESTED** [review]
needs to use existing tab control impl.

**Comment by EVAST9919** [comment]
> Why is this not using `OsuTabControl`, which already implements this tab design? We shouldn't be implementing local variants.
> 
> See `HeaderTabControl` for instance. It's also used at song select and other places.

I didn't use OsuTabControl because it has different behavior. I mean, hover/hoverlost/onclick

**Comment by peppy** [comment]
Different in what way? Basically we have a limited number of UI controls which should be shared. If a UI element differs, ask first because it's usually a remnant of flyte making new designs for every single UI implementation which is totally wrong and bad and we don't do it any more.

**Comment by EVAST9919** [comment]
For example: Hovering item OsuTabControl cause box resize and there's no colour change, and in web implementation hovering item cause just colour change.
Also using enums in OsuTabControl means every enum item will be added to the control, but in our case we need only part of them.

**Comment by peppy** [comment]
As above: we're not here to implement many different UI designs.

- You can easily make the line different by exposing a `CreateLine()`  method
- You can add a ctor to get around the "don't need all items" issue.
- You can adopt whichever colour change version you feel feels better and use in both places.

**Review by peppy — APPROVED** [review]

**Pull request merged**